tests/unittests/CoreGraphics/CGContextTests.mm

```diff
@@ -419,3 +420,68 @@ virtual void SetUp() {
 
 INSTANTIATE_TEST_CASE_P(CGContextCoordinateSpace, ContextCoordinateTest, ::testing::ValuesIn(coordinateTestTuples));
 #endif
+
+TEST(CGContext, DrawAnImageIntoContext) {
+    woc::unique_cf<CGColorSpaceRef> rgbColorSpace(CGColorSpaceCreateDeviceRGB());
+
+    woc::unique_cf<CGContextRef> context(CGBitmapContextCreate(
+        nullptr, 512, 256, 8, 4 * 512 /* bytesPerRow = bytesPerPixel*width*/, rgbColorSpace.get(), kCGImageAlphaPremultipliedFirst));
+
+    CFDataRef data = (CFDataRef)[NSData dataWithContentsOfFile:getPathToFile(@"jpg1.jpg")];
+    woc::unique_cf<CGDataProviderRef> dataProvider(CGDataProviderCreateWithCFData(data));
+
+    woc::unique_cf<CGImageRef> cgImage(CGImageCreateWithJPEGDataProvider(dataProvider.get(), NULL, NO, kCGRenderingIntentDefault));
+    ASSERT_NE(cgImage, nullptr);
+
+    CGRect bounds = { 0, 0, 512, 256 };
+    CGAffineTransform flip = CGAffineTransformMakeScale(1, -1);
+    CGAffineTransform shift = CGAffineTransformTranslate(flip, 0, bounds.size.height * -1);
+    CGContextConcatCTM(context.get(), shift);
+
+    // data check before draw
+    BYTE* dataPtr = static_cast<BYTE*>(CGBitmapContextGetData(context.get()));
+    ASSERT_NE(dataPtr, nullptr);
+    EXPECT_EQ(dataPtr[0], 0xcd);
+
+    CGContextDrawImage(context.get(), bounds, cgImage.get());
+
+    // data check after draw
+    dataPtr = static_cast<BYTE*>(CGBitmapContextGetData(context.get()));
+    ASSERT_NE(dataPtr, nullptr);
+    EXPECT_EQ(dataPtr[0], 0x97);
```

Comment:
better option is more bit comparison.
But given it's a red rectangle that get's flooded. this should be fine for now.